source/extensions/stat_sinks/common/statsd/statsd.h

```diff
@@ -38,11 +41,12 @@ class Writer : public ThreadLocal::ThreadLocalObject {
 class UdpStatsdSink : public Stats::Sink {
 public:
   UdpStatsdSink(ThreadLocal::SlotAllocator& tls, Network::Address::InstanceConstSharedPtr address,
-                const bool use_tag);
+                const bool use_tag, const std::string& prefix = getDefaultPrefix());
   // For testing.
   UdpStatsdSink(ThreadLocal::SlotAllocator& tls, const std::shared_ptr<Writer>& writer,
-                const bool use_tag)
-      : tls_(tls.allocateSlot()), use_tag_(use_tag) {
+                const bool use_tag, const std::string& prefix = getDefaultPrefix())
+      : tls_(tls.allocateSlot()), use_tag_(use_tag),
+        prefix_(prefix.empty() ? getDefaultPrefix() : prefix) {
```

Comment:
Is there a reason we want to disallow an empty prefix since the user already has the option to not provide the argument and get the default? (same for TCP)

Reply:
An empty prefix doesn't make sense, imo. If an empty prefix is passed in, we just use the default instead. Also, if a prefix isn't specified `statsd_sink.prefix()` in `createStatsSink` will be an empty string. We can avoid multiple checks for an empty string by doing it here.

Reply:
SGTM - I didn't realize this was coming directly from the proto, so there is no way to specify a default.